example/src/domain/machinery/Form.js

```diff
@@ -0,0 +1,109 @@
+import { useFieldSnapshot, useFieldValue, useFormField, useNumberFormField, useArrayFormField } from '@kaliber/forms'
+import { Code } from './Code'
+
+export function FormValues({ form }) {
+  const formState = useFieldSnapshot(form)
+  return <Code value={formState} indent />
+}
+
+export function FormTextInput({ field, label }) {
+  const { name, state, eventHandlers } = useFormField(field)
+  return <InputBase type='text' {...{ name, label, state, eventHandlers }} />
+}
+
+export function FormNumberInput({ field, label }) {
+  const { name, state, eventHandlers } = useNumberFormField(field)
+  return <InputBase type='text' {...{ name, label, state, eventHandlers }} />
+}
+
+export function FormCheckbox({ field, label }) {
+  const { name, state, eventHandlers } = useBooleanFormField(field)
+  const { value } = state
+  console.log(`[${name}] render checkbox field`)
+  return (
+    <LabelAndError {...{ name, label, state }}>
+      <input
+        id={name}
+        type='checkbox'
+        checked={value || false}
+        {...eventHandlers}
+      />
+    </LabelAndError>
+  )
+}
+
+function useBooleanFormField(field) {
+  const { name, state, eventHandlers: { onChange, ...originalEventHandlers } } = useFormField(field)
+  const eventHandlers = { ...originalEventHandlers, onChange: e => onChange(e.target.checked) }
+  return { name, state, eventHandlers }
+}
+
+export function FormArrayField({ field, render, initialValue }) {
```

Comment:
De naam `FormArrayField` hier was wat verwarrend, omdat het component meer doet dan enkel een array aan values bijhouden (namelijk markup en gedrag toevoegen), ten opzichte van de FormObjectField en FormFieldValue die precies doen wat ze zeggen.

Reply:
Deze lijkt meer op `FormTextInput`, zit niet in de library maar in het example. Wat zou een betere naam zijn denk je?